tt-train/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.18)
+cmake_policy(VERSION 3.18)
+include(cmake/compilers.cmake)
+
+if(DEFINED ENV{CMAKE_C_COMPILER} AND DEFINED ENV{CMAKE_CXX_COMPILER})
+    message(STATUS "Setting C and C++ compiler from environment variables")
+    set(CMAKE_C_COMPILER $ENV{CMAKE_C_COMPILER})
+    set(CMAKE_CXX_COMPILER $ENV{CMAKE_CXX_COMPILER})
+endif()
+
+if(CMAKE_CXX_COMPILER AND CMAKE_C_COMPILER)
+    message(STATUS "Using specifed C++ compiler: ${CMAKE_CXX_COMPILER}")
+    message(STATUS "Using specifed C compiler: ${CMAKE_C_COMPILER}")
+else()
+    message(STATUS "No C or C++ compiler specified, defaulting to Clang-17")
+    FIND_AND_SET_CLANG17()
+endif()
+
+project(ml-framework-cpp)
+CHECK_COMPILERS()
+
+if("$ENV{TT_METAL_HOME}" STREQUAL "")
+    message(FATAL_ERROR "TT_METAL_HOME is not set")
+endif()
+if("$ENV{ARCH_NAME}" STREQUAL "")
+    message(FATAL_ERROR "ARCH_NAME is not set")
+endif()
+
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -DDEBUG=DEBUG")
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DDEBUG=DEBUG")
+set(CMAKE_CXX_FLAGS_CI "-O3 -DDEBUG=DEBUG")
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    find_library(LIBC++ c++)
+    find_library(LIBC++ABI c++abi)
+    if(NOT LIBC++ OR NOT LIBC++ABI)
+        message(
+            FATAL_ERROR
+            "libc++ or libc++abi not found. Make sure you have libc++ and libc++abi installed and in your PATH"
+        )
+    endif()
+    # making it global settings for now
+    add_compile_options(-stdlib=libc++)
+    add_link_options(-stdlib=libc++)
+endif()
+
+message(STATUS "c++ Standard: ${CMAKE_CXX_STANDARD}")
+
+include(cmake/dependencies.cmake)
+
+message($ENV{TT_METAL_HOME})
+
+message(STATUS "TT_METAL_HOME: $ENV{TT_METAL_HOME}")
+
+message($ENV{TT_METAL_HOME}/tt_metal/third_party/src/firmware/riscv/$ENV{ARCH_NAME})
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# 3rd party projects
+add_subdirectory(3rd_party/tokenizers-cpp)
+add_subdirectory(3rd_party/wandb-cpp)
+# ttml projects
+
+add_subdirectory(sources)
+include(CTest)
```

Comment:
This should come near after `project` and before we start processing subdirectories.  But it's also going to be inherited from the top-level, now.
There's a lot in this file that should be deleted.  This file will become just what's of interest to `tt-train` without all the boilerplate.

Reply:
Not sure I understand what you mean about CTest